mesonbuild/wrap/wrap.py

```diff
@@ -266,44 +276,38 @@ def get_git(self) -> None:
-                subprocess.check_call([git, 'submodule', 'update',
-                                       '--init', '--checkout', '--recursive', *depth_option],
-                                      cwd=self.dirname)
+                verbose_git(['submodule', 'update', '--init', '--checkout',
+                             '--recursive', *depth_option], self.dirname, check=True)
             push_url = self.wrap.values.get('push-url')
             if push_url:
-                subprocess.check_call([git, 'remote', 'set-url',
-                                       '--push', 'origin', push_url],
-                                      cwd=self.dirname)
+                verbose_git(['remote', 'set-url', '--push', 'origin', push_url], self.dirname, check=True)
         else:
             if not is_shallow:
-                subprocess.check_call([git, 'clone', self.wrap.get('url'),
-                                       self.directory], cwd=self.subdir_root)
+                verbose_git(['clone', self.wrap.get('url'), self.directory], self.subdir_root, check=True)
                 if revno.lower() != 'head':
-                    if subprocess.run([git, 'checkout', revno], cwd=self.dirname).returncode != 0:
-                        subprocess.check_call([git, 'fetch', self.wrap.get('url'), revno], cwd=self.dirname)
-                        subprocess.check_call([git, 'checkout', revno], cwd=self.dirname)
+                    if verbose_git(['checkout', revno], self.dirname):
+                        verbose_git(['fetch', self.wrap.get('url'), revno], self.dirname, check=True)
+                        verbose_git(['checkout', revno], self.dirname, check=True)
             else:
-                subprocess.check_call([git, 'clone', *depth_option,
-                                       '--branch', revno,
-                                       self.wrap.get('url'),
-                                       self.directory], cwd=self.subdir_root)
+                verbose_git(['clone', *depth_option, '--branch', revno, self.wrap.get('url'),
+                             self.directory], self.subdir_root, check=True)
             if self.wrap.values.get('clone-recursive', '').lower() == 'true':
-                subprocess.check_call([git, 'submodule', 'update',
-                                       '--init', '--checkout', '--recursive', *depth_option],
-                                      cwd=self.dirname)
+                verbose_git(['submodule', 'update', '--init', '--checkout', '--recursive', *depth_option],
+                            self.dirname, check=True)
+                # git submodule update --recursive calls git clone recursively,
+                # and on Windows it will undo the console mode we set in mlog
+                # and cause ANSI colors to stop working. Set it again.
+                if platform.system().lower() == 'windows':
+                    mlog._windows_ansi()
```

Comment:
Is this still needed now that you set stdin to DEVNULL? Why would it be needed here and not anywhere else?

Reply:
I clarified the comment, should make more sense now.

Reply:
Ok I see, now you moved it to the common function, makes sense.